samwise.js

```diff
@@ -1,67 +1,67 @@
 /**
```

Comment:
SCRIPT-I like how you've used promise chaining here, it makes the code flow very clear.